src/classes/routes/RouterFactory.ts

```diff
@@ -0,0 +1,60 @@
+import { Router, Request, Response, NextFunction } from 'express';
+import PostImageController from './PostImageController';
+import Logger from '../Logger';
+import ConfigurationManager from '../ConfigurationManager';
+import * as multer from 'multer';
+import * as cors from 'cors';
+import { ExpressMiddleware } from '../../types';
+
+class RouterFactory {
+
+    constructor (private configurationManager: ConfigurationManager, private logger: Logger) { }
+
+    public buildImageRouter (): Router {
+        const router: Router = Router();
+        const fileInputName = this.configurationManager.getFileInputName();
+        const uploadFilePath = this.configurationManager.getUploadPath();
+
+        router.use(this.createCorsMiddleware());
```

Comment:
I think you should move `const router: Router = Router();` to line 17. The router initialization is separated by 2 lines of unrelated code.
Or better have a function to initialize the router, something like this:
```
public buildImageRouter (): Router {
        const fileInputName = this.configurationManager.getFileInputName();
        const uploadFilePath = this.configurationManager.getUploadPath();
        const uploadMiddleware = this.createUploadMiddleware(fileInputName, uploadFilePath);
        const corsMiddleware  = this.createCorsMiddleware();
        return this.buildRouter(corsMiddleware , uploadMiddleware);
}
public buildRouter (corsMiddleware: ExpressMiddleware, uploadMiddleware: ExpressMiddleware): Router {
        const router: Router = Router();
        router.use(corsMiddleware);
        router.options('*', corsMiddleware);
        router.post('/image', uploadMiddleware, (request: Request, response: Response, next: NextFunction) => {
            const controller = new PostImageController(this.logger);
            controller.executeController(request, response, next);
        });
        return router;
}
```